### PR TITLE
Add API to flag a review

### LINKED ISSUE #3344: Add API to allow users to flag an add-on review
Supports #3024

### PR DESCRIPTION
**Pull request by diox**
Fix #3344

### CODE REVIEW

File: docs/topics/api/reviews.rst [hunk 1]
```diff
@@ -133,4 +133,37 @@ This endpoint allows you to reply to an existing user review.
...
+
+
+----
+Flag
+----
+
+.. review-flag:
+
+This endpoint allows you to flag an existing user review, to let an editor know
+that something is wrong with it.
```
Comment (EnTeQuAk) [inline docs/topics/api/reviews.rst]: "may be wrong with it" sounds less dramatic

File: docs/topics/api/reviews.rst [hunk 2]
```diff
@@ -133,4 +133,37 @@ This endpoint allows you to reply to an existing user review.
...
+
+ .. note::
+     Requires authentication and a user account different from the one that
+     posted the review.
+
+.. http:post:: /api/v3/addons/addon/(int:id|string:slug|string:guid)/reviews/(int:id)/flag/
+
+    :<json string flag: A :ref:`constant<review-flag-constants>` describing the reason behind the flagging.
+    :<json string|null note: A note to explain further the reason behind the flagging.
```
Comment (EnTeQuAk) [inline docs/topics/api/reviews.rst]: We should maybe specify whether or not the note is required, e.g in case of "other"

**Comment by EnTeQuAk** [comment]
r+wc

File: docs/topics/api/reviews.rst (hunk 2 above)
Comment (eviljeff) [inline docs/topics/api/reviews.rst]: Is this enforced in the API - do you get an error back if you supply a flag of `reveiw_flag_reason_spam` and a note also?  If not, should it?

Comment (diox) [inline docs/topics/api/reviews.rst]: If you supply a note, it automatically changes the flag to `review_flag_reason_other`, as noted in the docs (and tested).

I'll add a test and docs regarding `note` being required for when you do specify the `review_flag_reason_other` flag.

Comment (EnTeQuAk) [inline docs/topics/api/reviews.rst]: > as noted in the docs (and tested)

I must have missed that part, sorry

**Pull request merged**